docs/build-provenance-for-all-registries.md

```diff
@@ -0,0 +1,104 @@
+# Build Provenance for All Registries
+
+There’s a new security capability that registries can offer - build provenance - which verifiably links a package back to its source code and build instructions.
+
+Build provenance starts with the build system making information about source code and build instructions available in a non-falsifiable way. During a build, that information is sent to a public-good signing service called [Sigstore](https://www.sigstore.dev/how-it-works), which provides a public transparency log for auditing purposes as well as enabling signing without requiring individual users to manage keys. The signed build provenance is then sent to a registry on package upload, and served to users when they consume a package. In this document, we describe the design decisions and high-level implementation for adding build provenance to npm, to show other registries how they could also support this capability.
+
+Before we talk about the implementation, let's talk through the design decisions. **The goal is to have packages be transparent about their contents.** Note that we aren’t saying a package with provenance is _inherently_ secure or trustworthy, just that non-falsifiable links to source code and build instructions exist for consumers to follow and make their own determination, either through manual inspection or by evaluating provenance with their own policies. This is a stronger guarantee than information optionally and manually being supplied by maintainers.
```

Comment:
It feels like it would help to have two additional items in this introduction:

1. A system diagram indicating what components are affected, and how they interoperate with existing components.  (e.g. code added in: build/publish CLI, package registry API, package fetch CLI.  Used but not changed: Sigstore, package storage)

2. A brief explainer of the properties added to the system, probably as a bulleted list.  i.e. "Packages _can_ include signed, verified descriptions of how they were built", "Package provenance is associated with an identity that attests to it", "End-users can make decisions based on package provenance", "Signed provenance _does not_ guarantee that a package has no vulnerabilities, or that it is non-malicious".

Reply:
I moved the component diagram up and added a "How Build Provenance Works" section to the introduction for your first suggestion.

I like your second suggestion as well, but I think that will make more sense in a follow-up "why Sigstore for build provenance" doc, that will also address other feedback on this pull request.